modules/services/batsignal.nix

```
@@ -0,0 +1,181 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.batsignal;
+
+  commandLine = concatStringsSep " " ([ "${cfg.package}/bin/batsignal" ]
+    ++ optional (cfg.warningLevelPercent != null)
+    "-w ${toString cfg.warningLevelPercent}"
+    ++ optional (cfg.criticalLevelPercent != null)
+    "-c ${toString cfg.criticalLevelPercent}"
+    ++ optional (cfg.criticalLevelPercent != null)
+    "-d ${toString cfg.dangerLevelPercent}"
+    ++ optional (cfg.fullLevelPercent != null)
+    "-f ${toString cfg.fullLevelPercent}"
+    ++ optional (cfg.warningLevelMessage != null)
+    "-W '${cfg.warningLevelMessage}'"
+    ++ optional (cfg.criticalLevelMessage != null)
+    "-C '${cfg.criticalLevelMessage}'"
+    ++ optional (cfg.dangerLevelCommand != null) (let
+      cmd = pkgs.writeShellScript "batsignal-dangercmd" cfg.dangerLevelCommand;
+    in "-D ${cmd}")
+    ++ optional (cfg.fullLevelMessage != null) "-F '${cfg.fullLevelMessage}'"
+    ++ optional (cfg.batteryNames != null)
+    "-n '${concatStringsSep "," cfg.batteryNames}'"
+    ++ optional (cfg.updateIntervalSeconds != null)
+    "-m ${toString cfg.updateIntervalSeconds}"
+    ++ optional (cfg.appName != null) "-a '${cfg.appName}'"
+    ++ optional (cfg.icon != null) "-I ${cfg.icon}");
+
+in {
+  meta.maintainers = with maintainers; [ loicreynier ];
+
+  options.services.batsignal = {
+    enable = mkEnableOption "batsignal battery monitor daemon";
+    package = mkPackageOption pkgs "batsignal" { };
+
+    warningLevelPercent = mkOption {
+      type = types.nullOr (types.ints.between 0 100);
+      default = null;
+      example = 15;
+      description = ''
+        Warning level percentage of the battery capacity.
+        0 disables this level.
+      '';
+    };
+
+    criticalLevelPercent = mkOption {
+      type = types.nullOr (types.ints.between 0 100);
+      default = null;
+      example = 5;
+      description = ''
+        Critical level percentage of the battery capacity.
+        0 disables this level.
+      '';
+    };
+
+    dangerLevelPercent = mkOption {
+      type = types.nullOr (types.ints.between 0 100);
+      default = null;
+      example = 2;
+      description = ''
+        Danger level percentage of the battery capacity.
+        0 disables this level.
+      '';
+    };
+
+    fullLevelPercent = mkOption {
+      type = types.nullOr (types.ints.between 0 100);
+      default = null;
+      example = 0;
+      description = ''
+        Full level percentage of the battery capacity.
+        0 disables this level.
+      '';
+    };
+
+    warningLevelMessage = mkOption {
+      type = types.nullOr types.lines;
+      default = null;
+      example = literalExpression ''
+        Battery low
+      '';
+      description = "Warning level message.";
```

I think this lines up a bit better with the man page: https://github.com/electrickite/batsignal/blob/master/batsignal.1.in#L46

(also do the same for all of the other `*LevelMessage` descriptions)

```suggestion
      description = "The message to show when the battery reaches the warning level.";
```